### ISSUE TITLE
[FEATURE] AI: Local LLM Training Infrastructure

### ISSUE THREAD

**Issue by lukassherman27** [lukassherman27/Benlsey-Operating-System#484]
## Overview
Build infrastructure to collect training data, fine-tune models on Bensley content, and deploy locally for privacy, speed, and unlimited context.

## User Story
As **Bensley**, I want a private, fine-tuned AI that understands our design philosophy, terminology, and approach without relying on external APIs for every query.

## Current State
- Vision.md mentions Phase 4 (2027) for Local AI
- `training_data` table exists but unused
- No Ollama integration
- No fine-tuning pipeline
- All AI calls go to cloud APIs

### Must Have (P0)
- [ ] Training data collection from approved suggestions
- [ ] Training data export in standard format (JSONL)
- [ ] Ollama installation guide
- [ ] Basic local model integration (Llama 3)
- [ ] Tiered routing: local for simple, cloud for complex

### Should Have (P1)
- [ ] LoRA fine-tuning on Bensley data
- [ ] Terminology recognition (project codes, design terms)
- [ ] Philosophy-aware responses
- [ ] Cost tracking: local vs cloud

### Nice to Have (P2)
- [ ] Multiple specialized models (content, proposals, design)
- [ ] Embedding model for semantic search
- [ ] Hardware recommendations document
- [ ] Model evaluation benchmarks

### Database Changes
Existing `training_data` table is sufficient.

Additional tracking:
```sql
-- Track model deployments
CREATE TABLE local_models (
    model_id INTEGER PRIMARY KEY AUTOINCREMENT,
    model_name TEXT NOT NULL,
    base_model TEXT, -- llama3, mistral, etc
    fine_tuned INTEGER DEFAULT 0,
    training_samples INTEGER,
    training_date TEXT,
    version TEXT,
    ollama_model_name TEXT,
    is_active INTEGER DEFAULT 0,
    performance_score REAL,
    notes TEXT
);

-- Track API usage for cost comparison
CREATE TABLE ai_usage_log (
    log_id INTEGER PRIMARY KEY AUTOINCREMENT,
    timestamp TEXT DEFAULT (datetime('now')),
    provider TEXT, -- openai, anthropic, ollama
    model TEXT,
    prompt_tokens INTEGER,
    completion_tokens INTEGER,
    cost_usd REAL,
    task_type TEXT,
    success INTEGER DEFAULT 1
);
```

### Training Data Format
```json
{"prompt": "Draft an Instagram caption for Shinta Mani Wild showing the jungle", "response": "The jungle doesn't ask permission. It simply grows, breathes, lives – and we designed around it. 🌿 #ShintaManiWild #JungleFirst #BensleyDesign", "source": "human_approved", "task": "instagram_caption"}
```

### API Endpoints
| Method | Endpoint | Purpose |
|--------|----------|---------|
| GET | /api/ai/training-data | Export training data |
| POST | /api/ai/training-data/add | Add approved sample |
| GET | /api/ai/models | List available models |
| POST | /api/ai/models/switch | Change active model |
| GET | /api/ai/usage | Usage statistics |

### Services
```
backend/services/local_ai/
├── training_collector.py    # Collect approved outputs
├── training_exporter.py     # Export to JSONL
├── ollama_client.py         # Local model API
├── model_router.py          # Route to local/cloud
├── cost_tracker.py          # Usage tracking
└── fine_tuner.py            # LoRA fine-tuning script
```

### Hardware Requirements
**Minimum (inference only):**
- Mac M1 with 16GB RAM
- Or Linux with 16GB RAM + GPU

**Recommended (fine-tuning):**
- Mac M2/M3 with 64GB RAM
- Or Linux with 32GB RAM + RTX 4090

## Data Requirements
**Collect over time:**
- [ ] 1,000+ approved AI suggestions
- [ ] 500+ approved content drafts
- [ ] 200+ approved proposal sections
- [ ] 100+ Bill quote transcriptions

## Dependencies
- Depends on: #477 (AI Content Generation - generates training data)
- Depends on: #483 (Brain Chat - generates training data)

## Estimation
- Backend: 3 days (infrastructure)
- Training: 2-5 days (iterative)
- Evaluation: 1 day
- Total: 6-9 days

## Acceptance Criteria
- [ ] Training data export works
- [ ] Ollama runs locally with Llama 3
- [ ] Simple queries routed to local model
- [ ] Cost savings measurable vs cloud
- [ ] 10,000+ training samples before fine-tuning

## Labels
`area/ai`, `type/feature`, `priority:p2`, `phase:4-future`
